### PR TITLE
always set a default name in the qgis project to avoid discrepancies with GetCapabilities

### PR DESCRIPTION
**Pull request by Gustry**
## Description

always set a default name in the qgis project to avoid discrepancies with GetCapabilities

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [x] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

**Review by rldhont — APPROVED** [review]
Thanks!

**Comment by Gustry** [comment]
Ok, Travis is happy now. Review are welcome. Thanks @rldhont

**Comment by rldhont** [comment]
@pblottiere @sbrunner can you give you opinion ?

**Comment by pblottiere** [comment]
Nice work @Gustry! 

 It is just a thought, but do you think it could be possible to set the current style of `QgsMapLayerStyleManager` to the default one without explicitly passing the "default" string in parameter? (which is by the way a translated string)

File: src/core/qgsmaplayerstylemanager.h [hunk 1]
```diff
@@ -161,6 +161,7 @@ class CORE_EXPORT QgsMapLayerStyleManager : public QObject
     QMap<QString, QgsMapLayerStyle> mStyles;
     QString mCurrentStyle;
     QgsMapLayerStyle *mOverriddenOriginalStyle = nullptr;
+    QString mEmptyStyleName = tr( "default" );
```
Comment (m-kuhn) [inline src/core/qgsmaplayerstylemanager.h]: I think this should go to a private method instead.

**Review by m-kuhn — CHANGES_REQUESTED** [review]
I'm somehow undecided on the use of localized strings that are saved to the project file, translation should be considered representation not data. Although I don't expect issues in the case here. Just to keep in mind ☺

File: src/core/qgsmaplayerstylemanager.h (hunk 1 above)
Comment (Gustry) [inline src/core/qgsmaplayerstylemanager.h]: Done

**Comment by Gustry** [comment]
I understand about the translated string, but users can already set their own style name if they rename it. Should we remove the use of the `tr` function?

File: src/core/qgsmaplayerstylemanager.h (hunk 1 above)
Comment (m-kuhn) [inline src/core/qgsmaplayerstylemanager.h]: Can you adjust the name without a `m`-prefix?

Comment (Gustry) [inline src/core/qgsmaplayerstylemanager.h]: I did that, the name of the function is `mDefaultStyleName`, not good?

Comment (Gustry) [inline src/core/qgsmaplayerstylemanager.h]: SOrry, I read too quick ;-)
I thought that every private members/methods should have a `m` prefix. I will fix that

**Comment by m-kuhn** [comment]
I'm not completely sure here, just wondering what we will do if we go for https://github.com/qgis/QGIS-Enhancement-Proposals/issues/90 or a multilingual server. However, it's no blocker for me, since the user can adjust it manually.

File: src/core/qgsmaplayerstylemanager.h (hunk 1 above)
Comment (m-kuhn) [inline src/core/qgsmaplayerstylemanager.h]: Just in case -> `Ctrl + Shift + R` will rename it quickly for you ;)

Comment (Gustry) [inline src/core/qgsmaplayerstylemanager.h]: Fixed with `cmd + shift + R` ;-)

File: tests/src/core/testqgsmaplayerstylemanager.cpp [hunk 2]
```diff
@@ -74,7 +74,7 @@ void TestQgsMapLayerStyleManager::testDefault()
   QVERIFY( mgr );
 
   QCOMPARE( mgr->styles().count(), 1 );
-  QCOMPARE( mgr->style( QString() ).isValid(), true );
+  QCOMPARE( mgr->style( QString( "default" ) ).isValid(), true );
```
Comment (m-kuhn) [inline tests/src/core/testqgsmaplayerstylemanager.cpp]: That should be a `QStringLiteral`

File: src/core/qgsmaplayerstylemanager.h [hunk 3]
```diff
@@ -161,6 +161,7 @@ class CORE_EXPORT QgsMapLayerStyleManager : public QObject
     QMap<QString, QgsMapLayerStyle> mStyles;
     QString mCurrentStyle;
     QgsMapLayerStyle *mOverriddenOriginalStyle = nullptr;
+    QString defaultStyleName();
```
Comment (m-kuhn) [inline src/core/qgsmaplayerstylemanager.h]: This method should be `const`

**Comment by rldhont** [comment]
@m-kuhn, are the changes good for you ?

**Comment by m-kuhn** [comment]
Yes, good job

**Pull request merged**